UI.Business/CustomElements/BasicElement.cs

```diff
@@ -0,0 +1,98 @@
+            return element.Text;
+        }
+        public bool Enabled()
+        {
+            _logger.LogDebug("Element is enabled");
+            return element.Enabled;
+        }
+
+        public int Height()
+        {
+            _logger.LogDebug($"Elements height is {element.Size.Height}");
+            return element.Size.Height;
+        }
+
+        public int Width()
+        {
+            _logger.LogDebug($"Elements width is {element.Size.Width}");
+            return element.Size.Width;
+        }
+
+        public void ScrollToElementJavaScript()
+        {
+            _logger.LogDebug("Scroll element into view");
+            JSExecutor().ExecuteScript("arguments[0].scrollIntoView(true);", element);
+        }
+
+        public bool IsElementScrolledIntoViewJavaScript()
+        {
+            bool isScrolledIntoView = (bool)JSExecutor().ExecuteScript("var elem = arguments[0],                 " +
+                                                                     "  box = elem.getBoundingClientRect(),    " +
+                                                                     "  cx = box.left + box.width / 2,         " +
+                                                                     "  cy = box.top + box.height / 2,         " +
+                                                                     "  e = document.elementFromPoint(cx, cy); " +
+                                                                     "for (; e; e = e.parentElement) {         " +
+                                                                     "  if (e === elem)                        " +
+                                                                     "    return true;                         " +
+                                                                     "}                                        " +
+                                                                     "return false;                            ", element);
+            return isScrolledIntoView;
+        }
+
+        public void DragAndDrop(IWebElement targetElement)
+        {
+            Actions actions = new Actions(Driver());
+
+            actions.ClickAndHold(element)
+                .MoveToElement(targetElement)
+                .Release()
+                .Build()
+                .Perform();
+        }
+
+        public bool Disabled()
+        {
+            _logger.LogDebug("Element is disabled");
+            return !element.Enabled;
+        }
+
+        public void Click()
+        {
+            _logger.LogDebug("Click on element");
+            element.Click();
+        }
+
+        public IJavaScriptExecutor JSExecutor()
+        {
+            IJavaScriptExecutor js = (IJavaScriptExecutor)((IWrapsDriver)element).WrappedDriver;
+            return js;
+        }
+
+        public IWebDriver Driver()
+        {
+            IWebDriver driver = ((IWrapsDriver)element).WrappedDriver;
+            return driver;
+        }
+
```

Comment:
Maybe it is better to use protected variables in class instead of public methods. It seems to me that this will be more reliable, since when creating a class object it will be impossible to obtain an instance of the driver. The driver instance will be available only to this class and its descendants. What do you think about it?

Reply:
Good point

Reply:
Done